File: utils/batch/consumer-fixtures/maven-jdbc-runtime/src/main/java/issue731/consumer/Consumer.java
package issue731.consumer;

import io.bluetape4k.batch.CheckpointJson;
import io.bluetape4k.batch.jdbc.ExposedJdbcBatchJobRepository;

public final class Consumer {
    private Consumer() {}

    public static Class<?>[] compileProbe() {
        return new Class<?>[] {CheckpointJson.class, ExposedJdbcBatchJobRepository.class};
    }

    public static void main(String[] args) {
        if (compileProbe().length != 2) {
            throw new IllegalStateException("JDBC runtime probe did not load both public types");
        }
    }
}

File: utils/batch/consumer-fixtures/jdbc-runtime/src/main/kotlin/issue731/consumer/Consumer.kt
package issue731.consumer

import io.bluetape4k.batch.CheckpointJson
import io.bluetape4k.batch.jdbc.ExposedJdbcBatchJobRepository

fun compileProbe(): List<Class<*>> = listOf(
    CheckpointJson::class.java,
    ExposedJdbcBatchJobRepository::class.java,
)
